spec/forms/waste_carriers_engine/base_forms_spec.rb

```diff
@@ -46,6 +46,7 @@ module WasteCarriersEngine
         context "when the resource is not a transient_registration" do
```

Comment:
Is this accurate? A NewRegistration is still a type of TransientRegistration.

Reply:
yes, this is not testing the validity of a `new_regisration` object in the base_form, but the validity of a `registration` object passed as a transient registration, so is still correct. You could argue that there is no test coverage for when we have a transient registration that is of type new registration, but as we discussed during the dev meeting we aren't sure about keeping this validation at all, and I will have to sort it out for validating the correct CBDU/CBDL numbers are generated later on so I just added that `is_a?` as a placeholder and make things pass now. We will need to go in more details about that in some other PR/meeting :D

Reply:
So the first `allow` is making sure that `is_a?(TransientRegistration)` always returns false, but the second `allow` makes sure we still actually call `is_a?(NewRegistration)` and use the real response – am I understanding this right?

Reply:
No, the second call is stubbing `is_a?(NewRegistration)` and making it return `nil`